### ISSUE TITLE
Add filter in webhook proxy to prevent unwanted builds

### ISSUE THREAD

**Issue by villanlu** [opendevstack/ods-core#487]
While migrating existing projects to ODS some branches may still not have a correct Jenkinsfile, this led pods to CrashLoopBackOff and therefore unnecessarily consume resources. It would be great to specify which branches you want to actually build. For example, I could specify the webhook proxy to only build "develop", "qa" and "master" branches.

**Comment by clemensutschig** [comment]
@michaelsauter  - I thought this is already there (remember the MRO discussion we had - on release branches and [ci skip])? 

I can't find it in the code or doc though
https://www.opendevstack.org/ods-documentation/opendevstack/3.x/jenkins/webhook-proxy.html

**Comment by michaelsauter** [comment]
No, the webhook proxy does not allow to filter branches. You can only filter events.

The pipeline can filter in Jenkins via `[ci skip]`, but it does already start a Jenkins Pipeline.

To be honest, I don't see enough value yet to build this:

* The webhook proxy is per project, not per component. Would it make sense to configure branches on a project level?
* The issue seems to be during migration only. Why not simply commit a `Jenkinsfile` into the branches which don't have it yet? If you don't commit into the branches at all, then no build is triggered. If you commit, why not add a `Jenkinsfile` as well?

**Comment by clemensutschig** [comment]
@villanlu  - lucas - ok to close?

**Comment by villanlu** [comment]
@clemensutschig If you don't see enough value, you can close it. But I'd keep it open at least with low priority. I could eventually add the feature myself to the webproxy GO files and create the PR.
For now, we are managing it with a new variable: "buildBranches" in the Jenkinsfile, which then is used in the ci skip. All done in our devops-jenkins-shared-library fork.
